doajtest/unit/test_article_match.py

```diff
@@ -172,8 +173,51 @@ def test_04_with_doi_instead(self):
         assert len(l) == 2
         l.sort()
         assert expected == l
+
+    def test_05_full_doi(self):
+        """ Test that we still detect duplicate DOIs when we have the full URI, not just the 10. """
+        # make ourselves a couple of example articles
+        a = models.Article()
+        b = a.bibjson()
+        b.title = "Example A article with a DOI"
+        b.add_identifier('doi', "https://doi.org/10.doi")
+        a.save()
```

Comment:
I did give that a go, but it was causing intermittent failures on my machine when we check we retrieve the latest - it seems the timestamps aren't dissimilar enough when performing blocking saves.

Reply:
I may revisit the ```differentiate``` arg to enforce that

Reply:
```blocking=True``` can't really be used here to ensure timestamps are sufficiently different. That's because if we are saving 2 new objects they don't have a last_updated date which is being intentionally modified, so the difference check is never run. I think currently the only way to enforce that is to actually wait the time. Alternatively we could pass in an offset to the ```save()``` method for the purpose of testing, or even better, wrap the save function in our testing framework to allow this without editing the DAO's save method.

Reply:
And for the record, ```differentiate``` is used to ensure our saves have different timestamps but not block.

Reply:
Ah, ok, I don't think it's important enough to look into any further, I missed the point about making sure the timestamps were sufficiently different.